### PR TITLE
deprecated Twig_Environment::computeAlternatives()

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

File: lib/Twig/Error/Syntax.php [hunk 1]
```diff
@@ -17,4 +17,38 @@
...
+        }
+
+        $this->rawMessage .= sprintf('. Did you mean "%s"?', implode('", "', $alternatives));
+        $this->updateRepr();
+    }
+
+    /**
+     * @internal
+     *
+     * To be merged with the addAlternatives() method in 2.0.
```
Comment (stof) [inline lib/Twig/Error/Syntax.php]: shouldn't this be ``addMessageSuggestions`` ?

File: lib/Twig/Error/Syntax.php [hunk 2]
```diff
@@ -17,4 +17,38 @@
...
+     * @param string $name  The original name of the item that does not exist
+     * @param array  $items An array of possible items
+     */
+    public function addMessageSuggestions($name, array $items)
+    {
+        if (!$alternatives = self::computeAlternatives($name, $items)) {
+            return;
+        }
+
+        $this->rawMessage .= sprintf('. Did you mean "%s"?', implode('", "', $alternatives));
```
Comment (stof) [inline lib/Twig/Error/Syntax.php]: should we really add the dot here ? It means that the message without alternatives does not have a final dot

File: lib/Twig/Error/Syntax.php (hunk 1 above)
Comment (fabpot) [inline lib/Twig/Error/Syntax.php]: fixed

File: lib/Twig/Error/Syntax.php (hunk 2 above)
Comment (fabpot) [inline lib/Twig/Error/Syntax.php]: fixed

Comment (fabpot) [inline lib/Twig/Error/Syntax.php]: I've also added some new tests as it was not covered.

**Comment by stof** [comment]
:+1:

**Pull request merged**